src/components/BTable/BTable.vue

```diff
@@ -0,0 +1,320 @@
+      type: Number,
+      required: false
+    },
+    /**
+    * The number of rows to be displayed in the table per page. Defualts to 10 if not provided.
+    */
+    rowsPerPage: {
+      type: Number
+    },
+    /**
+    * The total number of rows for the table.
+    */
+    totalResults: {
+      type: Number,
+      required: true
+    },
+    /**
+    * The different row sizes availablt to the user, will default to [10, 20, 50] if not provided.
+    */
+    rowDisplayOptions: {
+      type: Array,
+    }
+  },
+  data () {
+    return {
+      rows: this.buildRows(this.tableData, this.headers),
+      unsorted: require('../../../static/unSortedArrows.svg'),
+      currentPage: this.pageNumber ? this.pageNumber : 0,
+      options: this.rowDisplayOptions ? this.rowDisplayOptions : [10, 20, 50],
+      currentRowsPerPage: this.rowsPerPage ? this.rowsPerPage : 10,
+      sorted: {},
+      id: null
+    }
+  },
+  mounted() {
+    this.id = this._uid
+  },
+  computed: {
+    /**
+    * @returns{String}
+    * This builds the string used in the table to show meta information on the table rows.
+    */
+    results: function () {
+      const start = (this.currentPage * this.currentRowsPerPage) + 1
+      let end = (start + parseInt(this.currentRowsPerPage)) - 1
+      end = end > this.totalResults ? this.totalResults : end
+      return `${start}-${end} of ${this.totalResults}`
+    }
+  },
+  methods: {
+    /**
+    * @param {Array<Object|String>} rowData
+    * @param {Array<String>} headers
+    * @returns {Array<String>}
+    * This maps the rowData with the applicable headers to form the rows.
+    * If the render key is present, a supplied render funtion will be called
+    * to populate that cell instead of a string.
+    */
+    buildRows: function(rowData, headers) {
+      let rows = []
+      rowData.forEach( function(data) {
+        let row = []
+        headers.forEach( function(header) {
+          if (header.render) {
+            row.push(header.render(data[header.key]))
+          } else {
+            row.push(data[header.key])
+          }
+        })
+        rows.push(row)
+      })
+      return rows
+    },
+    /**
+    * @param {String} sortBy
+    * Populates an object to track sort state and emits
+    * that object to the 'sort' event which a parent
+    * component can listen to using '.$on'
+    */
+    onColumnHeaderClick: function(sortBy) {
+      if (sortBy in this.sorted) {
+        this.sorted[sortBy] === 'desc' ? this.sorted[sortBy] = 'asc' : delete this.sorted[sortBy]
+      } else {
+        this.sorted[sortBy] = 'desc'
+      }
+      this.$emit("sort", this.sorted)
+      this.$forceUpdate()
+    },
+    /**
+    * @param {String} page
+    * Updates the current row the table is on, and emits
+    * an object holding the new page and the rowzsize
+    * to the 'paginate' event which a parent
+    * component can listen to using '.$on'
+    */
+    onPaginateClick: function(page) {
```

Comment:
I changed tableData to:
```
let tableData = []
for (i=0; i < 100; i++) {
  tableData.push({
    header_one: "column " + i, 
    header_two: {name: "column " + i*i, id: i+1}, 
    header_three: {source:'https://raw.githubusercontent.com/fanappics/buoy/master/buoylogo.png', name: 'buoy logo'}, 
    header_four: 'column 4', 
    header_five: 'column a', 
    header_six: 'column c', 
    name: "example row " + i, 
    actions: [(i%2 ? 'delete' : 'edit')]
  })
} 
```

and it did not paginate:

![1000rows](https://user-images.githubusercontent.com/18075411/36623995-625eca78-18c7-11e8-9f48-980c52cda82b.gif)

Reply:
it won't paginate like that.  Those arrows fire off events for the user to hook into.  Like sorting, pagination is handled by the user.

Reply:
gotcha

Reply:
@mlear Is that documented in the styleguide?

Reply:
its documented in the code for the methods, but currently we are not including the methods in the styleguide... I'll update the markdown to mention this specifically